diffrax/_solver/foster_srk.py

```diff
@@ -0,0 +1,52 @@
+import numpy as np
+
+from .srk import AbstractSRK, StochasticButcherTableau
+
+
+x1 = (3 - np.sqrt(3)) / 6
+x2 = np.sqrt(3) / 3
+
+tab = StochasticButcherTableau(
+    c=np.array([0.5, 1.0]),
+    b_sol=np.array([x1, x2, x1]),
+    a=[np.array([0.5]), np.array([0.0, 1.0])],
+    aW=[np.array([0.5]), np.array([0.0, 1.0])],
+    aH=[np.array([np.sqrt(3)]), np.array([0.0, 0.0])],
+    bW=np.array([x1, x2, x1]),
+    bH=np.array([1.0, 0.0, -1.0]),
+    b_error=np.array([x1 - 0.5, x2, x1 - 0.5]),
+    bW_error=np.array([x1 - 0.5, x2, x1 - 0.5]),
+    bH_error=np.array([1.0, 0.0, -1.0]),
+    additive_noise=False,
+)
+
+
+class FosterSRK(AbstractSRK):
+    r"""A Stochastic Runge-Kutta method based on equation Definition $1.6$ from
+
+    ??? cite "Reference"
+
+        ```bibtex
+        @misc{foster2023convergence,
+            title={On the convergence of adaptive approximations
+            for stochastic differential equations},
+            author={James Foster},
+            year={2023},
+            eprint={2311.14201},
+            archivePrefix={arXiv},
+            primaryClass={math.NA}
+        }
+        ```
+
+    """
+
+    tableau = tab
+
+    def __init__(self):
+        super().__init__()
```

Comment:
Removing this works in practice, but pyright complains:
```
/home/andy/PycharmProjects/diffrax_STLA/test/test_integrate.py:259:26 - error: Argument missing for parameter "minimal_levy_area"
```

Reply:
Ah, right. So this is because Equinox Modules and dataclasses have slightly different rules for `__init__` methods and inheritance. (Equinox Modules inherit `__init__` methods, but dataclasses create a fresh `__init__` method in the subclass.)

So I think the best way to fix this is to remove the `minimal_levy_area` field, which is a little odd anyway -- as it's a deterministic function of the tableau, I think it'd be better to turn into a method or `property`, and to just remove both field and `__init__` method from `AbstractSRK`.

Reply:
That's actually a very good idea, I changed it into a method.

I cannot get rid of the ` __init__` though, because then when running `diffeqsolve` I get an error saying that `numpy.ndarray` is unhashable. I think that error message is very badly written. If I didn't already know that the only change I made was deleting ` __init__`, then there's no way it could point me to the right fix. Not sure if the JAX error itself should be written better, or if you should do a catch in equinox, but in any case, something more informative would be useful.

So yes, now I have the following in `srk.py`:
```python
def __init__(self):
    pass
```

Reply:
I believe this is because you have declared the tableau as an attribute of the instance, not an attribute of the class. Right now you have
```python
tableau: StochasticButcherTableau
```
when you want
```python
tableau: AbstractClassVar[StochasticButcherTableau]
```
(see also `AbstractRungeKutta`).

The error is because when it tries to hash the instance, it tries to hash all its attributes (more precisely, the dataclass fields).